### PR TITLE
feat: sandbox preloads by default

### PR DESCRIPTION
**Pull request by nornagon**
BREAKING CHANGE

#### Description of Change
Ref https://github.com/electron/electron/issues/28466#issuecomment-887009418, #32868

NB. breaking change is already documented in breaking-changes.md: https://github.com/electron/electron/blob/main/docs/breaking-changes.md#default-changed-renderers-without-nodeintegration-true-are-sandboxed-by-default

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Renderers are now sandboxed by default unless `nodeIntegration: true` or `sandbox: false` is specified.

### CODE REVIEW

File: shell/browser/web_contents_preferences.cc [hunk 1]
```diff
@@ -315,10 +315,9 @@ bool WebContentsPreferences::GetPreloadPath(base::FilePath* path) const {
 bool WebContentsPreferences::IsSandboxed() const {
   if (sandbox_)
     return *sandbox_;
-  bool sandbox_disabled_by_default =
-      node_integration_ || node_integration_in_worker_ || preload_path_ ||
-      !SessionPreferences::GetValidPreloads(web_contents_->GetBrowserContext())
-           .empty();
+  bool sandbox_disabled_by_default = node_integration_ ||
+                                     node_integration_in_worker_ ||
+                                     node_integration_in_sub_frames_;
```
Comment (miniak) [inline shell/browser/web_contents_preferences.cc:320]: why also `node_integration_in_sub_frames_`?

**Review by jkleinsc — APPROVED** [review]
API LGTM

**Comment by nornagon** [comment]
Hm... that test is failing because it uses `nodeIntegrationInSubframes: true` without `nodeIntegration: true`, so the test gets sandboxed, which somehow seems to break the preload script? Even though the preload script just exposes `ipcRenderer`.

I'm not quite sure what's going on here, it seems to me like that test ought to work unmodified.

File: spec/webview-spec.js [hunk 2]
```diff
@@ -243,6 +243,7 @@ describe('<webview> tag', function () {
     it('preload script can require modules that still use "process" and "Buffer" when nodeintegration is off', async () => {
       const message = await startLoadingWebViewAndWaitForMessage(webview, {
         preload: `${fixtures}/module/preload-node-off-wrapper.js`,
+        nodeIntegration: true,
```
Comment (miniak) [inline spec/webview-spec.js:246]: why not `sandbox: false`?

Comment (nornagon) [reply spec/webview-spec.js:246]: 🤷🏻 sure, seems about the same to me but i'll switch it if you like.

**Review by miniak — APPROVED** [review]

**Review by itsananderson — APPROVED** [review]
API LGTM

**Comment by nornagon** [comment]
Mac failures are flake; merging.

**Pull request merged**